### DISCUSSION TITLE
File Input doesn't have an option to modify the displayed file name (To User)

### DISCUSSION THREAD

**Discussion by samnayakawadi** [saadeghi/daisyui#2102] [Ideas / feature requests]
I have used this File Input in my react project where I am facing this issue.

Issue
- Whenever I go to another component (onClick route change), The file input is reset.
- This shows the 'No file chosen'
- But, The user has already chosen the file & navigated to other components. But sees this message when back to the File Input.
- Hence, there must be a tag that overrides the file name (Instead of relying on the default HTML input filename)

For reference see this bulma.io doc (https://bulma.io/documentation/form/file/#modifiers)
- Here, They have given a file-name class to update the name as per developers' needs.
I hope This is clear. Thanks

**Comment by saadeghi** [comment]
This is the default HTML element behavior and we can't really control it using CSS.  
If we visually show a text on top of the field for a file name (like the Bulma examples you linked) it won't function unless you manually change the file name using JS every time a file is chosen (as you can see this behavior on Bulma docs. It's just hardcoded text)  

Also it won't make any difference for your component. If no file is chosen by the user you should show "no file chosen" text anyway.  
It's an input for user to choose files and if your page routing or component re-rendering is emptying the field, why would you want to show another text? because the input is actually empty and won't upload anything.  

If user is choosing a file and you want to keep the value even after component re-rendering or page navigation, maybe saving the input value in local storage would help.

Anyway, I'm not sure how visually showing a value while the input is actually empty is going to be helpful but if you want that, you can do something like this: https://play.tailwindcss.com/tCjo9RXeol
